compiler/src/test/java/com/squareup/anvil/compiler/ReflectionTestUtils.kt

```diff
@@ -14,3 +15,14 @@ val Member.isAbstract: Boolean get() = Modifier.isAbstract(modifiers)
 fun <T : Any> Class<T>.createInstance(
   vararg initargs: Any?
 ): T = declaredConstructors.single().use { it.newInstance(*initargs) } as T
+
+inline fun <T, E : Executable> E.use(block: (E) -> T): T {
+  // Deprecated since Java 9, but many projects still use JDK 8 for compilation.
+  @Suppress("DEPRECATION")
+  val original = isAccessible
+
+  isAccessible = true
+  return block(this).also {
+    isAccessible = original
+  }
```

Comment:
If `block()` throws an exception, your `also` block won't be run, and the element will remain accessible. Using a try/finally usually works around this:

```kotlin
return try {
  isAccessible = true
  block(this)
} finally {
  isAccessible = original
}
```